graphql-dgs-pagination/src/main/kotlin/com/netflix/graphql/dgs/pagination/DgsPaginationTypeDefinitionRegistry.kt

```diff
@@ -42,7 +48,7 @@ class DgsPaginationTypeDefinitionRegistry {
                 definitions.add(createEdge(it.name))
             }
 
-        if (types.find { it.name == "PageInfo" } == null) {
+        if (types.find { it.hasDirective("connection") } != null && types.find { it.name == "PageInfo" } == null) {
```

Comment:
You can use `types.any(...)` instead of `.find`. It's the same, but returns a boolean, so it's a bit more readable.

Reply:
done.